lib/language_pack/ruby.rb

```diff
@@ -1236,7 +1009,6 @@ def load_bundler_cache
 
     old_rubygems_version = @metadata.read(ruby_version_cache).strip if @metadata.exists?(ruby_version_cache)
     old_stack = @metadata.read(stack_cache).strip if @metadata.exists?(stack_cache)
-    old_stack ||= DEFAULT_LEGACY_STACK
```

Comment:
Now that this default is gone, is it possible for this value to be `nil`? E.g.; could this line of code be reached and, if so, what would happen? 
```rb
puts "Purging Cache. Changing stack from #{old_stack} to #{@stack}"
```

Reply:
Good point.

The ruby buildpack has the concept of "metadata" that it stores between builds. This is durable information that is unfortunately stored in the cache. When this stack based cache clearing code was introduced, we were not persisting the state of the old stack in metadata for 100% of deployed apps, so if one was missing, we defaulted back to this value. 

Tthat is fixed (100% of deploys that have a prior deploy, have this metadata value). However it's worth checking that the first deploy doesn't mistakenly say that it's clearing the cache. I'll need to validate that and get back to you.